src/app/modules/core/advertisement/advertisement.service.ts

```diff
@@ -1,29 +1,67 @@
+import { HttpClient } from '@angular/common/http'
 import { Injectable } from '@angular/core'
-import {
-  MOCKADVERTISEMENTS_MOCK_DATA,
-  MockAdvertisement,
-} from '@modules/core/advertisement/mock-advertisement/mock-advertisement'
+import { MockAdvertisement } from '@modules/core/advertisement/mock-advertisement/mock-advertisement'
 import { Observable, of } from 'rxjs'
+import { flatMap } from 'rxjs/internal/operators'
+import { catchError, map, skipWhile, take, toArray } from 'rxjs/operators'
 
 @Injectable({
   providedIn: 'root',
 })
 export class AdvertisementService {
-  advertisements = MOCKADVERTISEMENTS_MOCK_DATA.map((advertisement) =>
-    MockAdvertisement.Build(advertisement)
-  )
+  advertisementsUrl = 'api/advertisements'
 
-  constructor() {}
+  private handleError<T>(
+    operation = 'operation',
+    result?: T
+  ): (error: any) => Observable<T> {
+    return (error: any): Observable<T> => {
+      console.error(error)
+
+      console.log(`AdvertisementService ${operation} failed ${error.message}`)
+
+      return of(result as T)
+    }
+  }
```

Comment:
Piuttosto che usare questo tipo di error handling e anche per gestire in maniera simile gli errori tra i vari servizi che stiamo creando potresti usare quello che avevo già scritto per il [reservation service](https://github.com/pc-codingavision/palermo-fe-final-project/blob/56e7cc07d04116d1dfd06fbd3d88d876fd2d566c/src/app/modules/shared/services/reservation/reservation.service.ts) 

Così avremo un `console.error` più specifico e un'errore più generico da mostrare tramite la `snackBar` all'utente
Ah e secondo me potresti spostare tutto questo a fine file dopo gli altri metodi `private` che definisci

Reply:
Non vanno lasciati console.log nel codice.

Reply:
In effetti, non ci avevo pensato. Grazie, Ludovico.